homeassistant/components/demo/notify.py

```diff
@@ -1,38 +1,45 @@
-"""Demo notification service."""
+"""Demo notification entity."""
 
 from __future__ import annotations
 
-from typing import Any
-
-from homeassistant.components.notify import BaseNotificationService
+from homeassistant.components.notify import DOMAIN, NotifyEntity
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 EVENT_NOTIFY = "notify"
 
 
-def get_service(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> BaseNotificationService:
-    """Get the demo notification service."""
-    return DemoNotificationService(hass)
-
-
-class DemoNotificationService(BaseNotificationService):
-    """Implement demo notification service."""
-
-    def __init__(self, hass: HomeAssistant) -> None:
-        """Initialize the service."""
-        self.hass = hass
-
-    @property
-    def targets(self) -> dict[str, str]:
-        """Return a dictionary of registered targets."""
-        return {"test target name": "test target id"}
-
-    def send_message(self, message: str = "", **kwargs: Any) -> None:
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the demo entity platform."""
+    async_add_entities([DemoNotifyEntity(unique_id="notify", device_name="Notifier")])
+
+
+class DemoNotifyEntity(NotifyEntity):
+    """Implement demo notification platform."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
```

Comment:
Shouldn't this be the default on `NotifyEntity` ?

Reply:
It is the default:
https://github.com/home-assistant/core/blob/fb5fc136e88335dab88425280b930e141deb11e0/homeassistant/components/notify/__init__.py#L131